cmd/metalctl/app/move.go

```diff
@@ -0,0 +1,347 @@
+	move.Flags().StringVar(&targetKubeconfig, "target-kubeconfig", "", "Kubeconfig pointing to the target cluster")
+	move.Flags().BoolVar(&crdsOnly, "crds-only", false, "migrate only the CRDs without CRs")
+	move.Flags().BoolVar(&crsOnly, "crs-only", false, "migrate only the CRs without CRDs")
+	move.Flags().StringVar(&namespace, "namespace", "", "namespace to filter CRDs and CRs to migrate. Defaults to all namespaces if not specified")
+	move.Flags().BoolVar(&dryRun, "dry-run", false, "show what would be moved without executing the migration")
+	move.Flags().BoolVar(&verbose, "verbose", false, "enable verbose logging for detailed output during migration")
+	move.MarkFlagRequired("source-kubeconfig")
+	move.MarkFlagRequired("target-kubeconfig")
+
+	if verbose {
+		slog.SetLogLoggerLevel(slog.LevelDebug)
+	}
+	return move
+}
+
+type Clients struct {
+	source client.Client
+	target client.Client
+}
+
+func makeClient(kubeconfig string) (client.Client, error) {
+	cfg, err := clientcmd.BuildConfigFromFlags("", kubeconfig)
+	if err != nil {
+		return nil, fmt.Errorf("failed to load cluster kubeconfig: %w", err)
+	}
+	return client.New(cfg, client.Options{Scheme: scheme})
+}
+
+func makeClients() (Clients, error) {
+	var clients Clients
+	var err error
+
+	clients.source, err = makeClient(sourceKubeconfig)
+	if err != nil {
+		return clients, fmt.Errorf("failed to construct a source cluster client: %w", err)
+	}
+	clients.target, err = makeClient(targetKubeconfig)
+	if err != nil {
+		return clients, fmt.Errorf("failed to construct a target cluster client: %w", err)
+	}
+	return clients, nil
+}
+
+func getMetalCrds(ctx context.Context, cl client.Client) ([]*apiextensionsv1.CustomResourceDefinition, error) {
+	crds := &apiextensionsv1.CustomResourceDefinitionList{}
+	if err := cl.List(ctx, crds); err != nil {
+		return nil, fmt.Errorf("couldn't list CRDs: %w", err)
+	}
+
+	metalCrds := make([]*apiextensionsv1.CustomResourceDefinition, 0)
+	for _, crd := range crds.Items {
+		if crd.Spec.Group == metalv1alphav1.GroupVersion.Group {
+			metalCrds = append(metalCrds, &crd)
+		}
+	}
+
+	return metalCrds, nil
+}
+
+func getMetalCrs(ctx context.Context, cl client.Client, crds []*apiextensionsv1.CustomResourceDefinition) ([]*unstructured.Unstructured, error) {
+	crs := make([]*unstructured.Unstructured, 0)
+	for _, crd := range crds {
+		CRs := &unstructured.UnstructuredList{}
+		CRs.SetGroupVersionKind(schema.GroupVersionKind{Group: crd.Spec.Group, Version: crd.Spec.Versions[0].Name, Kind: crd.Spec.Names.Kind})
+
+		if err := cl.List(ctx, CRs, &client.ListOptions{Namespace: namespace}); err != nil {
+			return nil, fmt.Errorf("couldn't list CRs: %w", err)
+		}
+		for _, cr := range CRs.Items { // won't work with go version <1.22
+			crs = append(crs, &cr)
+		}
+	}
+
+	return crs, nil
+}
+
+func clearFields(obj client.Object) map[string]interface{} {
```

Comment:
`map[string]any` is more concise than `map[string]interface{}`. Given how often it appears a type alias may make sense, e.g. `type MapStrAny map[string]any`.

Reply:
Actually, type assertion compares exact type names:
```
interface conversion: interface {} is map[string]interface {}, not app.MapStrAny
```
More information: https://stackoverflow.com/questions/19577423/how-to-cast-to-a-type-alias-in-go

I changed `map[string]interface{}`with `map[string]any`.